include_gif/src/lib.rs

```diff
@@ -1,25 +1,103 @@
 extern crate proc_macro;
 
 use proc_macro::TokenStream;
+use std::collections::HashSet;
 use std::fs::File;
 use std::io::Write;
+use syn::{parse_macro_input, Ident, LitStr};
+
+enum BppFormat {
+    Bpp1 = 0,
+    Bpp2 = 1,
+    Bpp4 = 2,
+}
+
+enum GlyphType {
+    Bagl,
+    Nbgl,
+}
+
+enum Input {
+    FileNameOnly(LitStr),
+    FileNameAndType(LitStr, GlyphType),
+}
+
+impl syn::parse::Parse for Input {
+    fn parse(input: syn::parse::ParseStream) -> syn::Result<Self> {
+        // Parse the filename
+        let filename = input.parse::<LitStr>()?;
+        if input.is_empty() {
+            Ok(Input::FileNameOnly(filename))
+        } else {
+            // Parse comma separator (won't be used, just to skip it)
+            let _: syn::Token![,] = input.parse()?;
+            // Parse the glyph type
+            let glyph_type = input.parse::<Ident>()?;
+            match glyph_type.to_string().as_str() {
+                "BAGL" => Ok(Input::FileNameAndType(filename, GlyphType::Bagl)),
+                "NBGL" => Ok(Input::FileNameAndType(filename, GlyphType::Nbgl)),
+                _ => Err(syn::Error::new_spanned(glyph_type, "Invalid glyph type")),
+            }
+        }
+    }
+}
 
 #[proc_macro]
 pub fn include_gif(input: TokenStream) -> TokenStream {
-    let filename = syn::parse_macro_input!(input as syn::LitStr);
+    let input = parse_macro_input!(input as Input);
+    match input {
+        Input::FileNameOnly(filename) => {
+            // Default to Bagl if no type is specified.
```

Comment:
As far as I have understood, BAGL will be soon replaced by NBGL, even for Nano devices. So maybe default value should be set to NBGL ?

Reply:
Risky choice, we should probably discuss that in our weekly meeting. :thinking: 
What do you think @xchapron-ledger ?

Reply:
@yogh333 actually it's used in quite a few apps : https://github.com/search?q=org%3ALedgerHQ+include_gif+language%3ARust&type=code&l=Rust

Reply:
Let's discuss about that during our next meeting, but even for existing apps, the modification is minor and pretty straightforward and it could be more future proof to me.